ADR/0033-provisioning-test-resources.md

```diff
@@ -0,0 +1,277 @@
+  ClusterPools ---> |reference| cluster-pull-secret
+  ArgoCDOperator --> |installs| HelmCharts
+  ArgoCDOperator --> |watches| Applications
+  HiveOperator --> |watches| ClusterDeployments
+  HiveOperator --> |watches| ClusterClaims
+  HiveOperator --> |4. creates| kubeconfigs
+  install-configs --> |contain copy of| cluster-pull-secret
+  install-configs --> |contain copy of| cluster-ssh-key
+  HelmCharts --> |3. deploy| ClusterDeployments
+  HelmCharts --> |3. deploy| ClusterClaims
+  ApplicationSets --> |reference| HelmCharts
+  ClusterClaims --> |reference| ClusterPools
+  IntegrationPipeline ---> |1. create| ClusterTemplateInstances
+  IntegrationPipeline ---> |6. reads| cti-kubeconfig
+  ClaaSOperator --> |2. creates| Applications
+  ClaaSOperator --> |watches| ClusterTemplateInstances
+  ClaaSOperator .-> |copies| kubeconfigs
+  ClaaSOperator --> |5. creates/deletes| cti-kubeconfig
+```
+
+#### CaaS, Hive & Bring Your Own Credentials
+
+```mermaid
+flowchart TD
+  HelmCharts
+
+  subgraph cluster [Cluster]
+    subgraph userns [User Accessible Namespace]
+      ClusterTemplateInstances
+      ClusterDeployments
+      ClusterPools
+      ClusterClaims
+      Applications
+      subgraph Secrets
+        kubeconfig
+        install-configs
+        aws-creds
+        cluster-pull-secret
+        cluster-ssh-key
+      end
+    end
+
+    subgraph hive [Hive Namespace]
+      HiveOperator
+    end
+
+    subgraph cluster-aas [ClaaS Namespace]
+      ClaaSOperator
+      ArgoCDOperator
+      ClusterTemplates
+      ApplicationSets
+    end
+  end
+
+  ClusterTemplateInstances --> |reference w/releaseImage| ClusterTemplates
+  ClusterTemplates --> |reference| ApplicationSets
+  ClusterDeployments ---> |reference| install-configs
+  ClusterDeployments ---> |reference| aws-creds
+  ClusterDeployments ---> |reference| cluster-ssh-key
+  ClusterDeployments ---> |reference| cluster-pull-secret
+  ClusterPools ---> |reference| install-configs
+  ClusterPools ---> |reference| aws-creds
+  ClusterPools ---> |reference| cluster-ssh-key
+  ClusterPools ---> |reference| cluster-pull-secret
+  ArgoCDOperator --> |installs| HelmCharts
+  ArgoCDOperator --> |watches| Applications
+  HiveOperator --> |watches| ClusterDeployments
+  HiveOperator --> |watches| ClusterClaims
+  HiveOperator --> |4. creates| kubeconfig
+  install-configs --> |contain copy of| cluster-pull-secret
+  install-configs --> |contain copy of| cluster-ssh-key
+  HelmCharts --> |3. deploy| ClusterDeployments
+  HelmCharts --> |3. deploy| ClusterClaims
+  ApplicationSets --> |reference| HelmCharts
+  ClusterClaims --> |reference| ClusterPools
+  IntegrationPipeline ---> |1. creates| ClusterTemplateInstances
+  IntegrationPipeline ---> |5. reads| kubeconfig
+  ClaaSOperator --> |2. creates| Applications
+  ClaaSOperator --> |watches| ClusterTemplateInstances
+```
+
+### Scalability
+
+The [CaaS] Operator should scale well. It hands off most workloads to ArgoCD. Even so,
```

Comment:
How will CaaS scale? Will we need to deploy new clusters to manage the scaling after some point in time? Will this scaling be automated or will it be associated with the scale of the Konflux "workload clusters" that are running users' builds? I assume that these clusters would need to be deployed 1:1 with Konflux member clusters?

Reply:
I've included more detail about the scaling characteristics for the management cluster(s). As for CaaS itself, I haven't seen any benchmarking so it's probably unknown. I suspect the limiting factor will be either ArgoCD or the provisioning option in used (Hyershift required more resources on the management cluster).